projects/packages/jetpack-mu-wpcom/src/features/block-patterns/class-wpcom-block-patterns-from-api.php

```diff
@@ -46,53 +46,26 @@ public function register_patterns() {
 		$pattern_categories = array();
 		$block_patterns     = $this->get_patterns( $patterns_cache_key );
 
+		// Register categories from first pattern in each category.
 		foreach ( (array) $block_patterns as $pattern ) {
 			foreach ( (array) $pattern['categories'] as $slug => $category ) {
-				// Register categories from first pattern in each category.
-				if ( ! isset( $pattern_categories[ $slug ] ) ) {
+				// Skip categories that start with an underscore
+				$is_hidden_category = substr( $slug, 0, 1 ) === '_';
+
+				if ( ! isset( $pattern_categories[ $slug ] ) && ! $is_hidden_category ) {
 					$pattern_categories[ $slug ] = array(
 						'label'       => $category['title'],
 						'description' => $category['description'],
 					);
-				}
-			}
-		}
-
-		// Unregister existing categories so that we can insert them in the desired order (alphabetically).
-		$existing_categories = array();
-		foreach ( \WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered() as $existing_category ) {
-			$existing_categories[ $existing_category['name'] ] = $existing_category;
-			unregister_block_pattern_category( $existing_category['name'] );
-		}
-
-		// Existing categories are registered in Gutenberg or other plugins.
-		// We overwrite them with the categories from Dotcom patterns.
-		$pattern_categories = array_merge( $existing_categories, $pattern_categories );
 
-		// Order categories alphabetically by their label.
-		uasort(
-			$pattern_categories,
-			function ( $a, $b ) {
-				return strnatcasecmp( $a['label'], $b['label'] );
-			}
-		);
-
-		// Move the Featured category to be the first category.
-		if ( isset( $pattern_categories['featured'] ) ) {
```

Comment:
The featured category will be moved to the top in https://github.com/WordPress/gutenberg/issues/60210. For now, we cannot control that ordering.